src/hidra/receiver/asapo_transfer.py

```diff
@@ -0,0 +1,180 @@
+                except Exception as e:
+                    logger.error("Transmission does not succeed. {e}. File ignored".format(e=str(e)))
+
+    def stop(self):
+        logger.info("Runner is stopped.")
+        self.stop_run.set()
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                     description='Transfer files metadata to ASAPO')
+    parser.add_argument('--config_path', type=str, help='Path to config directory')
+    parser.add_argument(
+        'identifier', type=str,
+        help='Beamline and detector ID information separated by a single underscore')
+
+    args = vars(parser.parse_args())
+    config = construct_config(args.pop('config_path'), args.pop('identifier'))
+    logging.basicConfig(format="%(asctime)s %(module)s %(lineno)-6d %(levelname)-6s %(message)s",
+                        level=getattr(logging, config.log_level))
+    logger.info("Start Asapo transfer with parameters %s", config)
+
+    worker_args = dict(
+        endpoint=config.endpoint,
+        beamtime=config.beamtime,
+        token=config.token,
+        n_threads=config.n_threads,
+        file_regex=config.file_regex,
+        default_data_source=config.default_data_source,
+        timeout=config.timeout,
+        beamline=config.beamline,
+        start_file_idx=config.start_file_idx)
+
+    logger.info("Creating AsapoWorker with %s", worker_args)
+    asapo_worker = AsapoWorker(**worker_args)
+
+    logger.info(
+        "Creating Transfer instance type=%s signal_host=%s detector_id=%s use_log=True",
+        "STREAM_METADATA", config.signal_host, config.detector_id)
+    query = Transfer("STREAM_METADATA", config.signal_host, detector_id=config.detector_id,
+                     use_log=True)
+
+    asapo_transfer = AsapoTransfer(
+        asapo_worker=asapo_worker,
+        query=query,
+        target_host=config.target_host,
+        target_port=config.target_port,
+        target_dir=config.target_dir,
+        reconnect_timeout=config.reconnect_timeout)
+
+    signal.signal(signal.SIGINT, lambda s, f: asapo_transfer.stop())
+    signal.signal(signal.SIGTERM, lambda s, f: asapo_transfer.stop())
+
+    run_transfer(asapo_transfer, config.reconnect_timeout)
+
+
+def construct_config(config_path, identifier):
+
+    # Read config file
+    with open(f"{config_path}/asapo_transfer_{identifier}.yaml", "r") as f:
+        config = yaml.load(f.read(), Loader=yaml.FullLoader)
+
+    # parse identifier to extruct `beamline` and `detector_id`
+    beamline, detector_id = identifier.split("_", maxsplit=1)
+    config['beamline'] = beamline
+    config['detector_id'] = detector_id
+
+    if 'default_data_source' not in config:
+        config['default_data_source'] = detector_id
+        if ".desy.de" in detector_id:
+            config['default_data_source'] = detector_id[:detector_id.find(".desy.de")]
```

Comment:
Maybe I am paranoid, but better remove that substring only when it is at the end:
```
def remove_suffix(string, suffix):
    if string.endswith(suffix):
        return string[:-len(suffix)]
    return string
```

By the way, if using `find`, there is no need to first test `in`. That just doubles the work.

Reply:
I assume this is an answer to my side note? What I meant there is to do the following instead:
```
index = string.find(substring)
if index >= 0:
    # do something with index
```
No need to check `in` first.

Reply:
Changed.